src/category_theory/category/Twop.lean

```diff
@@ -0,0 +1,113 @@
+
+instance : has_coe_to_sort Twop Type* := ⟨X⟩
+
+attribute [protected] Twop.X
+
+/-- Turns a two-pointing into a two-pointed type. -/
+def of {X : Type*} (to_two_pointing : two_pointing X) : Twop := ⟨X, to_two_pointing⟩
+
+alias of ← two_pointing.Twop
+
+instance : inhabited Twop := ⟨of two_pointing.bool⟩
+
+/-- Turns a two-pointed type into a bipointed type, by forgetting that the pointed elements are
+distinct.-/
+def to_Bipointed (X : Twop) : Bipointed := X.to_two_pointing.to_prod.Bipointed
+
+instance large_category : large_category Twop := induced_category.category to_Bipointed
+
+instance concrete_category : concrete_category Twop :=
+induced_category.concrete_category to_Bipointed
+
+instance has_forget_to_Bipointed : has_forget₂ Twop Bipointed :=
+induced_category.has_forget₂ to_Bipointed
+
+/-- Swaps the pointed elements of a two-pointed type. `two_pointing.swap` as a functor. -/
+@[simps] def swap : Twop ⥤ Twop :=
+{ obj := λ X, ⟨X, X.to_two_pointing.swap⟩, map := λ X Y f, ⟨f.to_fun, f.map_snd, f.map_fst⟩ }
+
+/-- The equivalence between `Twop` and itself induced by `prod.swap` both ways. -/
+@[simps] def swap_equiv : Twop ≌ Twop :=
+equivalence.mk swap swap
+  (nat_iso.of_components (λ X, { hom := ⟨id, rfl, rfl⟩, inv := ⟨id, rfl, rfl⟩ }) $ λ X Y f, rfl)
+  (nat_iso.of_components (λ X, { hom := ⟨id, rfl, rfl⟩, inv := ⟨id, rfl, rfl⟩ }) $ λ X Y f, rfl)
+
+@[simp] lemma swap_equiv_symm : swap_equiv.symm = swap_equiv := rfl
+
+end Twop
+
+@[simp] lemma Twop_swap_comp_forget_to_Bipointed :
+  Twop.swap ⋙ forget₂ Twop Bipointed = forget₂ Twop Bipointed ⋙ Bipointed.swap := rfl
+
+/-- The functor from `Pointed` to `Twop` which adds a second point. -/
+def Pointed_to_Twop_fst : Pointed.{u} ⥤ Twop :=
+{ obj := λ X, ⟨option X, ⟨X.point, none⟩, some_ne_none _⟩,
+  map := λ X Y f, ⟨option.map f.to_fun, congr_arg _ f.map_point, rfl⟩,
+  map_id' := λ X, Bipointed.hom.ext _ _ option.map_id,
+  map_comp' := λ X Y Z f g, Bipointed.hom.ext _ _ (option.map_comp_map  _ _).symm }
+
+/-- The functor from `Pointed` to `Twop` which adds a first point. -/
+def Pointed_to_Twop_snd : Pointed.{u} ⥤ Twop :=
+{ obj := λ X, ⟨option X, ⟨none, X.point⟩, (some_ne_none _).symm⟩,
+  map := λ X Y f, ⟨option.map f.to_fun, rfl, congr_arg _ f.map_point⟩,
+  map_id' := λ X, Bipointed.hom.ext _ _ option.map_id,
+  map_comp' := λ X Y Z f g, Bipointed.hom.ext _ _ (option.map_comp_map  _ _).symm }
+
+@[simp] lemma Pointed_to_Twop_fst_comp_swap :
+  Pointed_to_Twop_fst ⋙ Twop.swap = Pointed_to_Twop_snd := rfl
+
+@[simp] lemma Pointed_to_Twop_snd_comp_swap :
+  Pointed_to_Twop_snd ⋙ Twop.swap = Pointed_to_Twop_fst := rfl
+
+@[simp] lemma Pointed_to_Twop_fst_comp_forget_to_Bipointed :
+  Pointed_to_Twop_fst ⋙ forget₂ Twop Bipointed = Pointed_to_Bipointed_fst := rfl
+
+@[simp] lemma Pointed_to_Twop_snd_comp_forget_to_Bipointed :
+  Pointed_to_Twop_snd ⋙ forget₂ Twop Bipointed = Pointed_to_Bipointed_snd := rfl
+
+/-- `Pointed_to_Twop_fst` is one of the two free functors. -/
```

Comment:
Can you be more specific here? Free functors where they exist will be unique, but the two defs you have given establish an adjunction where the second functor differs; so calling them both free functors is somewhat misleading.